modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/ComponentBasedMetadataCacheIdGenerator.java

```diff
@@ -10,30 +10,27 @@
 
 import org.mule.runtime.api.dsl.DslResolvingContext;
 import org.mule.runtime.ast.api.ComponentAst;
-import org.mule.runtime.config.api.dsl.model.DslElementModelFactory;
 import org.mule.runtime.config.internal.model.ComponentModel;
+import org.mule.runtime.core.internal.locator.ComponentLocator;
 import org.mule.runtime.core.internal.metadata.cache.MetadataCacheId;
 import org.mule.runtime.core.internal.metadata.cache.MetadataCacheIdGenerator;
-import org.mule.runtime.core.internal.metadata.cache.MetadataCacheIdGeneratorFactory.ComponentLocator;
 
 import java.util.Optional;
 
 /**
  * A {@link ComponentModel} based implementation of a {@link MetadataCacheIdGenerator}
  *
  * @since 4.1.4, 4.2.0
+ * @deprecated Use {@link ComponentAstBasedMetadataCacheIdGenerator} directly.
```

Comment:
why not remove this class too?

Reply:
tooling is using it

Reply:
But we should change it so that it uses the new implementation. Al least, let's create a jira ticket and add a comment for it.